homework/list.js

```diff
@@ -1,79 +1,92 @@
 'use strict';
 
-(function() {
+(function(exports) {
+	var TodoListManager = function() {
+		this._listNoteContent = [];
+		this._wrapper = document.querySelector('#note-list-wrapper');
+	};
 
-  var _listNoteContent = [];
-  var _wrapper = document.querySelector('#note-list-wrapper');
+	TodoListManager.prototype = {
+		start() {
+			this.fetchList(function(data) {		
+				this.updateList(data)
+				.then((function(){
+					this.drawList(); 
+				}).bind(this)).then((function(){
+					this.preloadFirstNote();
+				}).bind(this)).catch((function(error){
+				}).bind(this));			
+			}.bind(this));
+			
+			
+			window.addEventListener('click', function(event) {
+				this.onNoteOpen(event);
+			}.bind(this));
+		},
+		 
+		onNoteOpen(event) {
+			if (event.target.classList.contains('note-title')) {
+				var id = event.target.dataset.noteId;
+				var content = this._listNoteContent[id];
+				window.dispatchEvent(new CustomEvent('note-open',{ detail: content }));
+			};
+		},
+		
+		preloadFirstNote() {
+			if (this._listNoteContent.length !== 0) {
+				var content = this._listNoteContent[0];
+				window.dispatchEvent(new CustomEvent('note-open',{ detail: content }));
+			}
+			return Promise.resolve();
```

Comment:
Well even for the demo purpose you don't need to "Promisified" every method, although I know I once mentioned that we now have a final strategy to make every method as a Promise. However, the premise was you firstly know why you need that, so that you can explain the strategy for your reviewer if it's necessary.